api/envoy/type/runtime_percent.proto

```diff
@@ -0,0 +1,22 @@
+syntax = "proto3";
+
+package envoy.type;
+
+import "validate/validate.proto";
+import "gogoproto/gogo.proto";
+
+import "envoy/type/percent.proto";
+
+option (gogoproto.equal_all) = true;
+
+// [#protodoc-title: RuntimePercent]
+
+// Runtime derived FractionalPercent with defaults for when the numerator or denominator is not
+// specified via a runtime key.
+message RuntimeFractionalPercent {
+  // Default value if the runtime value's for the numerator/denominator keys are not available.
+  envoy.type.FractionalPercent default_value = 1 [(validate.rules).message.required = true];
+
+  // Runtime key for a YAML representation of a FractionalPercent.
+  string runtime_key = 2;
```

Comment:
Should this have a constraint for non-empty?

Reply:
I didn't include any constraints because I assumed an empty string is also the default value if the field is not included. In addition, the associated value is validated.

Either way works for me.. let me know what you prefer and I'll implement that.

Reply:
This is fine as is.